### PR TITLE
add vtk export of displacements

### LINKED ISSUE #170: Add VTK output plots
Looking for previous contributions:
- [x] July (no vtk related features) (https://github.com/ONSAS/ONSAS.jl/tree/2a6ea5950320e0eb121a9f4146d8dec2730a45d8)
- [x] August (no vtk releated features)
https://github.com/ONSAS/ONSAS.jl/tree/f54d01f3ee5c556969370003ea52c07c20bd5114/src
- [x] October (no vtk releated features)   
(https://github.com/ONSAS/ONSAS.jl/tree/e959d4c59a5afd020ed87bee61738b1760a59354/src)
 - [x] November (no vtk releated features)   
 https://github.com/ONSAS/ONSAS.jl/tree/50ebd3713af376d1309d85e721792c7ad965df84
 - [x] December (no vtk releated features)   
 https://github.com/ONSAS/ONSAS.jl/tree/3b4f08ced4a763e0977080d701e7d2ee69140e0e/src
  - [x]  January (no vtk releated features)   
https://github.com/ONSAS/ONSAS.jl/tree/f8ae83369335f241a2e5039a4cc47eba0a988185/src
 
Maybe it was in another ONSAS.jl version @jorgepz ?

### PR DESCRIPTION
**Pull request by jorgepz**
add vtk export feature for displacements at last time of analysis closes https://github.com/ONSAS/ONSAS.jl/issues/170

### CODE REVIEW

**Comment by mvanzulli** [comment]
Issues related #170. And maybe is worth it to merge first #417

File: src/Interfaces/VTK.jl [hunk 1]
```diff
@@ -7,9 +7,34 @@ using Reexport
...
+export write_vtks
 
 # Add code here.
 
+function node_coordinates_matrix(nodes::Vector{N}) where {dim,T,N<:Node{dim,T}}
+    nodes_coords_matrix = Matrix{Float64}(undef, (length(nodes), dim))
+    dict_nodes_enum = Dict{AbstractNode,Int}()
+    for (i, n) in enumerate(nodes)
+        nodes_coords_matrix[i, :] = collect(coordinates(n))
+        dict_nodes_enum[n] = i
```
Comment (jorgepz) [inline src/Interfaces/VTK.jl]: I guess this should be removed from here since these lines are now in https://github.com/ONSAS/ONSAS.jl/pull/417

Comment (mvanzulli) [reply src/Interfaces/VTK.jl]: Yep, if u update the branch the Meshes provides those methods.  If u have already implemented it this way, we can refactor the implementation in a different PR.

Comment (jorgepz) [reply src/Interfaces/VTK.jl]: done in [178ce1a](https://github.com/ONSAS/ONSAS.jl/pull/414/commits/178ce1abcfe211d33bb8ce4f89641d2dbc2fa234)

File: src/Interfaces/VTK.jl [hunk 2]
```diff
@@ -7,9 +7,31 @@ using Reexport
...
+    end
+
+    #=
+    s = states_sol.analysis.s
+    for n in nodes(s)
+        # Get the displacements for all coordinates at the last time point.
+        data = [last(c) for c in displacements(states_sol, n)]
+        vtk_point_data(filename, data, "Displacements")
+    end
+    =#
```
Comment (mforets) [inline src/Interfaces/VTK.jl]: remove?

File: src/Interfaces/VTK.jl [hunk 3]
```diff
@@ -7,9 +7,31 @@ using Reexport
...
 
 using ..Meshes
+using ..Nodes
+using ..StructuralAnalyses
 
 @reexport import WriteVTK: vtk_grid
+export write_vtks
 
-# Add code here.
+function write_vtks(states_sol, filename)
```
Comment (mforets) [inline src/Interfaces/VTK.jl]: why not overload `write_vtk` with our own solution struct?

Comment (mforets) [reply src/Interfaces/VTK.jl]: as per the variable name, just `sol` seems more natural

File: src/Interfaces/VTK.jl [hunk 4]
```diff
@@ -7,9 +7,31 @@ using Reexport
...
+    nodes_mat = node_matrix(states_sol.analysis.s.mesh)
```
Comment (mforets) [inline src/Interfaces/VTK.jl]: here you can use `mesh` or another getter function, same below for the elements

File: src/Interfaces/VTK.jl (hunk 3 above)
Comment (mforets) [inline src/Interfaces/VTK.jl]: while at this, maybe add a docstring mentioning the fields currently exported? (so displacements for now)

File: src/Interfaces/VTK.jl (hunk 2 above)
Comment (jorgepz) [reply src/Interfaces/VTK.jl]: done in [cfcd24c](https://github.com/ONSAS/ONSAS.jl/pull/414/commits/cfcd24ced1944a2f76fecff1ad4b4dcd46901a7a)

**Comment by jorgepz** [comment]
There is still missing the deformation of the nodes coordinates

File: src/Interfaces/VTK.jl (hunk 3 above)
Comment (jorgepz) [reply src/Interfaces/VTK.jl]: good idea!

Comment (mforets) [reply src/Interfaces/VTK.jl]: ie.

```julia
@reexport import WriteVTK: write_vtk

function write_vtk(sol::AbstractSolution, filename::String)

...
end
```

File: src/Interfaces/VTK.jl (hunk 4 above)
Comment (mforets) [reply src/Interfaces/VTK.jl]: ie. `mesh(sol)` or `mesh(structure(sol))`, whatever makes sense. but in general that's cleaner than accessing fields directly.

File: src/Interfaces/VTK.jl (hunk 3 above)
Comment (mforets) [reply src/Interfaces/VTK.jl]: ie.

```julia
"""
Generate a VTK file given a solution struct.
Currently only `displacements` are exported for each time point.
"""
function write_vtk(...) # see comment above
```

**Review by mforets — APPROVED** [review]

**Comment by jorgepz** [comment]
impecable!

**Pull request merged**